tests/test_associations.py

```diff
@@ -95,8 +95,7 @@ def test_update_widget_associations(self):
 
         self.assertEqual(4, len(original_associations))
 
-        self.client.update_widget_associations(
-            widget=self.form_widget,
+        self.form_widget.update_associations(
             readable_models=[],
             writable_models=self.frame_model.properties[2:3],
```

Comment:
here all instances of this model are associated to the property grid (eg wheel case). That results in an overassociated widget (the frontend only retrieves the given `propertyInstanceId`). This is not desired. Widgets need to be explicitly associated in case of the grid (or attachment) there is only a single association needed to a single instance. In the backend we will fill the full association object including widget_id, activity_id, part_model_id, part_instance_id, property_model_id, property_instance_id, writable

Reply:
Would it be useful to write a test for associating every type of widget there is (and needs associations)? This would perhaps allow us to understand better what each widget expects as input on the `update_associations`, and perhaps also explain it in the documentation.